app/models/pawn.rb

```diff
@@ -1,5 +1,76 @@
 class Pawn < Piece
 
+  def valid_move?(new_x, new_y)
+    on_board?(new_x, new_y) && passes_pawn_rules?(new_x, new_y)
+  end
+
+  def passes_pawn_rules?(x, y)
+    if two_squares_forward?(x, y)
+      return true
+    elsif one_square_forward?(x, y)
+      return true
+    elsif diagonal_move?(x, y)
+      return true
+    end
+
+    return false
+  end
+
+  def two_squares_forward?(x, y)
+    first_move_position? && not_backwards?(x, y) && (self.y_cord - y).abs == 2 && self.x_cord - x == 0
+  end
+
+  def one_square_forward?(x, y)
+    not_backwards?(x, y) && (self.y_cord - y).abs == 1 && self.x_cord - x == 0
+  end
+
+  def diagonal_move?(x, y)
+    diagonal_allowed?(x, y) && not_backwards?(x, y) && (self.y_cord - y).abs == 1 && (self.x_cord - x).abs == 1
+  end
+
+  #Checks that square is on board
+  def on_board?(x, y)
+    x >= 0 && x <= 7 && y >=0 && y <= 7
+  end
+
+  #Checks that pawn is on the originating square aka first_move_position depending on its color
+  def first_move_position?
+    if self.y_cord == 1 && self.color == 'white'
+      return true
+    elsif self.y_cord == 6 && self.color == 'black'
+      return true
+    else
+      return false
```

Comment:
I'm not sure, but maybe try to take out all the returns.  like the if statement will return true automatically if the statement is true.  Not sure for else.  But there's at least an extra else, return false.